protocols/v2/subprotocols/template-distribution/src/set_new_prev_hash.rs

```diff
@@ -8,33 +8,56 @@ use binary_sv2::{Deserialize, Serialize, U256};
 #[cfg(not(feature = "with_serde"))]
 use core::convert::TryInto;
 
-/// ## SetNewPrevHash (Server -> Client)
-/// Upon successful validation of a new best block, the server MUST immediately provide a
-/// SetNewPrevHash message. If a [NewWork] message has previously been sent with the
-/// [future_job] flag set, which is valid work based on the prev_hash contained in this message, the
-/// template_id field SHOULD be set to the job_id present in that NewTemplate message
-/// indicating the client MUST begin mining on that template as soon as possible.
+/// Message used by an upstream to indicate the latest block header hash to mine on.
+///
+/// Upon validating a new best block, the server **must** immediately send this message.
+///
+/// If a `NewMiningJob` or `NewExtendedMiningJob` message has previously been sent with the
+/// Future Job flag set, the [`SetNewPrevHash::template_id`] field **should** be set to the
+/// `job_id` present in that message.
```

Comment:
```suggestion
/// `template_id` present in that message.
```

I think this should be the right description, I just asked on [discord](https://discord.com/channels/950687892169195530/1019861139573714994/1308745616943353898) to be 100% sure